### PR TITLE
More coding patterns converted to IR

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Few more coding patterns found in stdlib converted from `Tree` to `IR`. 

### Checklist

Please include the following checklist in your PR:

- [x] All code conforms to the [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
  - All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 1]
```diff
@@ -557,8 +608,75 @@ public void testBalanceLowerCase() throws Exception {
...
+              new directory . list name_filter=name_filter recursive=recursive
+                  """);
+  }
+
+  @Test
+  @Ignore
+  public void testLambdaBody() throws Exception {
+    parseTest("""
+    list =
+        all_files.filter file->
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:673]: @kazcw & co. I am trying to execute `main = IO.println "Hello"` and as far as I can say this is the main blocker. I am not aware of any reasonable workaround - right now I just replace the body with a dummy identifier `no_body_for_lambda`, but I am not sure if that's going to be enough to run the _"Hello world"_ example by Monday.

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 2]
```diff
@@ -641,17 +632,40 @@ yield switch (op.codeRepr()) {
             } else {
               var rhs = translateExpression(app.getRhs(), nil(), insideTypeSignature, true);
               var lhs = translateExpression(app.getLhs(), insideTypeSignature);
+              if (moreArgs.isEmpty() && rhs instanceof IR$Application$Prefix first && lhs instanceof IR$Application$Prefix pref) {
+                IR.CallArgument callArgument = new IR$CallArgument$Specified(Option.empty(), pref, loc, meta(), diag());
+                var args = cons(callArgument, first.arguments());
+                yield first.copy(
+                  first.copy$default$1(),
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:639]: :)

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 3]
```diff
@@ -697,23 +711,32 @@ yield switch (identifier) {
...
-                meta(), diag()
-              ),
-              rhs,
+            var name = new IR$Name$Literal(
+              op.codeRepr(), true,
               getIdentifiedLocation(app),
               meta(), diag()
             );
+            var loc = getIdentifiedLocation(app);
+            yield lhs != null ? new IR$Application$Operator$Binary(
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:732]: I would generally prefer using ternary operator only when something is a oneliner

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 4]
```diff
@@ -1598,6 +1650,11 @@ IR.Pattern translatePattern(Tree block, List<IR.Pattern> fields) {
       case Tree.Number num -> {
         yield new IR$Pattern$Literal((IR.Literal) translateDecimalLiteral(num), getIdentifiedLocation(num), meta(), diag());
       }
+      case Tree.TypeAnnotated anno -> {
+        var type = buildNameOrQualifiedName(maybeManyParensed(anno.getType()));
+        var expr = buildNameOrQualifiedName(maybeManyParensed(anno.getExpression()));
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:1655]: build name from an expression? This is intentional?

**Review by hubertp — APPROVED** [review]

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java (hunk 1 above)
Comment (kazcw) [reply engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:673]: Thank you for the test case. I'll try to get this working today.

Comment (kazcw) [reply engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:673]: Fix is in #3795

**Pull request merged**